### PR TITLE
Add data_files support

### PR DESCRIPTION
**Pull request by blink1073**
Uses upcoming functionality in https://github.com/jupyter/notebook/pull/3116 to enable the JupyterLab server extension when installed.

### CODE REVIEW

**Comment by stonebig** [comment]
would be great in tomorrow beta

**Comment by blink1073** [comment]
Won't work until Notebook 5.3 is released, I'm just putting it in there so it will "just work" after that and we can update our install instructions.

**Comment by stonebig** [comment]
AFAIK, notebook-5.3 was said to be released "this week" by @gnestor, which means tomorrow in western calendar.

**Comment by bollwyvl** [comment]
While it is just one file, I've been wondering if we want to encourage files bound for an `/etc/jupyter/*.d` files inside a directory. Making `etc/jupyter/jupyter_notebook_config.d/jupyterlab.json` seems a bit heavy, but would make the line more consistent across repos, a la `('etc', NAME, 'etc')`. We'd want to be consistent with the automadocumentation, i.e. https://github.com/jupyter-widgets/widget-ts-cookiecutter/pull/9

File: setup.py [hunk 1]
```diff
@@ -32,7 +32,8 @@
 data_files_spec = [
     ('share/jupyter/lab/static', '%s/static' % NAME, '**'),
     ('share/jupyter/lab/schemas', '%s/schemas' % NAME, '**'),
-    ('share/jupyter/lab/themes', '%s/themes' % NAME, '**')
+    ('share/jupyter/lab/themes', '%s/themes' % NAME, '**'),
+    ('etc/jupyter/jupyter_notebook_config.d', NAME, 'jupyterlab.json')
```
Comment (bollwyvl) [inline setup.py]: maybe trailing comma for THE FUTURE 🚀

**Comment by blink1073** [comment]
Maybe `config/jupyter_notebook_config.d/jupyterlab.json`?

**Comment by stonebig** [comment]
Notebook-5.3 should (have ?) be tomorrow https://gitter.im/jupyter/notebook?at=5a4be0d90163b02810a4b989

**Comment by bollwyvl** [comment]
Yeah, `config` works! Lab shouldn't be configuring much else, but who
knows... But as an example, it's useful.

Just so long as it's not a valid '.d' directory itself in the root, as that
would get picked up when running in the repo because of the path precedence!

**Comment by blink1073** [comment]
`jupyter-config`?

**Comment by bollwyvl** [comment]
Sounds very "canonical"!

**Pull request merged**